src/ucp/core/ucp_request.c

```diff
@@ -438,6 +444,9 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_request_memory_reg,
     flags  = UCT_MD_MEM_ACCESS_RMA | uct_flags;
     switch (datatype & UCP_DATATYPE_CLASS_MASK) {
     case UCP_DATATYPE_CONTIG:
+        if (req_dbg->flags & UCP_REQUEST_FLAG_USER_MEMH) {
```

Comment:
looks like now request is passed not for only debug purposes. I'd rename it to just `req`

Reply:
done